Publish payg-credit and link handshake (IOT-192)

jjmilburn opened the pull request:
These resources are implemented in https://github.com/angaza/nexus-embedded/tree/master/nexus , and demonstrated here: https://www.youtube.com/watch?v=A_SJdJS7SMQ

Introduce the OpenAPI resource models into the registry.

Commit:
feat: Publish payg-credit resource model

This resource model is deployed in pilot integrations, publish the draft
for public access.

Commit:
feat: Publish link handshake resource

This is implemented in the reference Nexus Channel security firmware,
and should be in the public repository of models.

ricehornet commented on resource_type_registry.md:
```diff
@@ -6,4 +6,6 @@
 | 201   | angaza.com.nexus.core.drlc | [drlc.oas.yaml](resource_types/core/energy/201-drlc/redoc_wrapper.md) |
 | 202   | angaza.com.nexus.core.operatingmode | [operatingmode.oas.yaml](resource_types/core/202-operatingmode/redoc_wrapper.md) |
 | 301   | angaza.com.nexus.core.samplelog | [samplelog.oas.yaml](resource_types/core/301-samplelog/redoc_wrapper.md) |
+| 401   | angaza.com.nexus.core.paygcredit | [paygcredit.oas.yaml](resource_types/core/401-paygcredit/redoc_wrapper.md) |
+| 1001  | angaza.com.nexus.link.handshake | [linkhandshake.oas.yaml](resource_types/link/1001-linkhandshake/redoc_wrapper.md) |
```
FYI we are using different RTRs for these resources in the embedded dev repo.

jjmilburn replied:
Updating on embedded side.

jjmilburn replied:
see - https://github.com/angaza/nexus-keycode-embedded-internal/pull/140

ricehornet commented on resource_types/core/401-paygcredit/paygcredit.oas.yaml:
```diff
@@ -0,0 +1,254 @@
+          x-example:
+            description: "x-example included only for automatic code generation."
+            rt:
+              - angaza.com.nexus.channel.core.paygcredit
```
```suggestion
              - angaza.com.nx.pc
```

This is what it is in the embedded dev repo.

ricehornet commented on resource_types/core/401-paygcredit/paygcredit.oas.yaml:
```diff
@@ -0,0 +1,254 @@
+          schema:
+              properties:
+                mo:
+                  $ref: '#/definitions/FullPAYGState/properties/mo'
```
I think it does not now actually do this (send `mo` with the error response)?

jjmilburn replied:
Will delete, to add as a later addition if needed.

jjmilburn replied:
Updated to remove this

ricehornet commented on resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml:
```diff
@@ -0,0 +1,285 @@
+            Current Channel Link handshake status
+          x-example:
+            rt: 
+              - angaza.com.nexus.link.handshake
```
```suggestion
              - angaza.com.nx.ln.hs
```

ricehornet commented on resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml:
```diff
@@ -0,0 +1,285 @@
+      cM:
+        description: >
+          "Challenge Mode".
+          Nexus Channel Link Handshake challenge authentication types supported. These
```
```suggestion
          Nexus Channel Link Handshake challenge authentication type for this handshake. These
```

ricehornet commented on resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml:
```diff
@@ -0,0 +1,285 @@
+      lS:
+        description: >
+          "Supported Link Security Modes".
+          Nexus Channel Link security modes supported by this device. These determine
```
```suggestion
          Nexus Channel Link security mode for this link. These determine
```

ricehornet commented on resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml:
```diff
@@ -0,0 +1,285 @@
+          (Only reported if `oic.if.baseline` interface query is supported)
+        items:
+          enum:
+          - angaza.com.nexus.link.handshake
```
```suggestion
          - angaza.com.nx.ln.hs
```

ricehornet reviewed (approved):
@jjmilburn LGTM, just have a few minor requests to improve clarity and sync with the embedded repo.

Commit:
fix: Correct docstrings, rt strings

Commit addressing PR review comments to correct incorrect documentation in the resource models for PAYG credit and link handshake

Co-authored-by: Eric <erict@angazadesign.com>

Commit:
fix: Remove 405 from PAYG credit

Not actually implemented in live systems.

Commit:
fix: Reconcile 'rt' string values with registry

Use shorter values to reduce transmitted string length, if used.
